test/e2e/e2e_test.go

```diff
@@ -323,6 +323,110 @@ func testE2E() {
 		}).Should(Succeed())
 	})
 
+	It("should react to failing devices with VolumeConditionAbnormal", func(ctx SpecContext) {
+		storageClass := "topolvm-provisioner-volumehealth"
+		By(fmt.Sprintf("deploying Pod with PVC based on StorageClass: %s", storageClass))
+		claimYAML := []byte(fmt.Sprintf(pvcTemplateYAML, "topo-pvc", "Filesystem", 200, storageClass))
+		podYaml := []byte(fmt.Sprintf(podVolumeMountTemplateYAML, "ubuntu", "topo-pvc"))
+
+		_, err := kubectlWithInput(claimYAML, "apply", "-n", ns, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred())
+		_, err = kubectlWithInput(podYaml, "apply", "-n", ns, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		By("confirming that the lv correspond to LogicalVolume resource is registered in LVM")
+		Eventually(func(g Gomega) error {
+			var pvc corev1.PersistentVolumeClaim
+			if err = getObjects(&pvc, "pvc", "-n", ns, "topo-pvc"); err != nil {
+				return err
+			}
+			return checkLVIsRegisteredInLVM(pvc.Spec.VolumeName)
+		}).Should(Succeed())
+
+		By("confirming the PVC is bound")
+		Eventually(func() error {
+			var pvc corev1.PersistentVolumeClaim
+			err := getObjects(&pvc, "pvc", "-n", ns, "topo-pvc")
+			if err != nil {
+				return fmt.Errorf("failed to get PVC. err: %w", err)
+			}
+			if pvc.Status.Phase != corev1.ClaimBound {
+				return errors.New("PVC is not bound")
+			}
+			return nil
+		}).Should(Succeed())
+
+		By("confirming that the specified device is mounted in the Pod")
+		Eventually(func() error {
+			return verifyMountExists(ns, "ubuntu", "/test1")
+		}).Should(Succeed())
+
+		By("triggering a volume health partial activation failure")
+		var failureDeviceCount int
+		if nonControlPlaneNodeCount == 0 {
+			failureDeviceCount = 1
+		} else {
+			failureDeviceCount = nonControlPlaneNodeCount
+		}
+		for i := 0; i < failureDeviceCount; i++ {
+			out, err := execAtLocal("sudo",
+				nil,
+				"dmsetup",
+				"remove",
+				"-f",
+				// This is the device name that is used in the volume health test vg for the crypt setup
+				fmt.Sprintf("/dev/mapper/crypt-%v", i+1),
+			)
+			if err != nil {
+				Expect(err.Error()).Should(ContainSubstring(fmt.Sprintf(
+					"remove ioctl on crypt-%v  failed: Device or resource busy", i+1)),
+					"The only accepted error for the dmsetup remove command is a busy device "+
+						"due to removing it while it is active")
+			}
+			if len(out) > 0 {
+				GinkgoT().Log(string(out))
+			}
```

Comment:
Is this for debugging? If so, could you format it to be more readable?
e.g. `GinkgoT().Logf("dmsetup stdout=%s", string(out))`